libs/common/src/change-stream/test-helpers/with-delete-many.api-tests.ts

```
@@ -0,0 +1,196 @@
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    afterAll(async () => {
+      await mongoDbClient.close();
+      deleteManyMock.mockRestore();
+      insertManyMock.mockRestore();
+    });
+
+    if (CHANGE_STREAM_ENABLED) {
+      describe('when the service is working normally', () => {
+        it('should add a deletion audit log', async () => {
+          await makeRequest();
+
+          const deletionAuditLogs = await deletionAuditLogsCollection
+            .find({
+              $or: getDeletedDocumentIds().map((deletedDocumentId) => ({
+                collectionName: { $eq: collectionName },
+                deletedDocumentId: { $eq: deletedDocumentId },
+              })),
+            })
+            .toArray();
+
+          expect(deletionAuditLogs).toEqual(
+            getDeletedDocumentIds().map((id) => ({
+              _id: expect.any(ObjectId) as ObjectId,
+              collectionName,
+              deletedDocumentId: new ObjectId(id),
+              auditRecord,
+              expireAt: expect.any(Date) as Date,
+            })),
+          );
+        });
+
+        it('should delete the documents', async () => {
+          await makeRequest();
+
+          const collection = await mongoDbClient.getCollection(collectionName);
+          const deletedDocuments = await collection
+            .find({
+              $or: getDeletedDocumentIds().map((deletedDocumentId) => ({
+                _id: { $eq: deletedDocumentId },
+              })),
+            })
+            .toArray();
+
+          expect(deletedDocuments.length).toBe(0);
+        });
+      });
+
+      describe('when deleting the document is not acknowledged', () => {
+        beforeEach(() => {
+          when(deleteManyMock)
+            // @ts-ignore
+            .calledWith(expect.anything() as object, { session: expect.any(ClientSession) as ClientSession })
+            .mockImplementationOnce(() => ({
+              acknowledged: false,
+            }));
```

This is potentially really pedantic, but I think that these mock implementations for `deleteMany` and `insertMany` should return promises that resolve/reject to these values, rather than returning them directly. I don't think it matters too much as `await`ing a non-async function has no effect, but I think it would nice to have as if we removed our `await`s from our source code then I think these tests would still pass when they should fail.